src/validators.ts

```diff
@@ -0,0 +1,135 @@
+import { Spec, ValidatorSpec } from './types'
+
+// Simplified adaptation of https://github.com/validatorjs/validator.js/blob/master/src/lib/isFQDN.js
+const isFQDN = (input: string) => {
+  if (!input.length) return false
+  const parts = input.split('.')
+  for (let part, i = 0; i < parts.length; i++) {
+    part = parts[i]
+    if (!/^[a-z\u00a1-\uffff0-9-]+$/i.test(part)) return false
+    if (/[\uff01-\uff5e]/.test(part)) return false // disallow full-width chars
+    if (part[0] === '-' || part[part.length - 1] === '-') return false
+  }
+  return true
+}
+
+// "best effort" regex-based IP address check
+// If you want a more exhaustive check, create your own custom validator, perhaps wrapping this
+// implementation (the source of the ipv4 regex below): https://github.com/validatorjs/validator.js/blob/master/src/lib/isIP.js
+const ipv4Regex = /^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$/
+const ipv6Regex = /([a-f0-9:]+:+)+[a-f0-9]+/
+const isIP = (input: string) => {
+  if (!input.length) return false
+  return ipv4Regex.test(input) || ipv6Regex.test(input)
+}
+
+const EMAIL_REGEX = /^[^@\s]+@[^@\s]+\.[^@\s]+$/ // intentionally non-exhaustive
+
+export class EnvError extends TypeError {
+  constructor(message?: string) {
+    super(message)
+    Error.captureStackTrace(this, EnvError)
+    this.name = 'EnvError'
+  }
+}
+
+export class EnvMissingError extends ReferenceError {
+  constructor(message?: string) {
+    super(message)
+    Error.captureStackTrace(this, EnvMissingError)
+    this.name = 'EnvMissingError'
+  }
+}
+
+export const makeValidator = <T>(parseFn: (input: string) => T, type: string = 'unknown') => {
```

Comment:
I don't get the point of the `type`-parameter and also it makes the code hard-ish to type when it's a big fn block and then a little var afterwards

Reply:
Haha, it was added so long ago and I can't actually see if it's even used anymore :D I think it may have been used for the default reporter output before. @SimenB maybe you remember? I agree we should kill it if it's no longer serving any purpose

Reply:
OK just removed this entirely in the latest commit 👍